Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM node:20.10.0-alpine3.19
+
+RUN apk -U add --no-cache bash git
+
+# override some config defaults with values that will work better for docker
+ENV ME_CONFIG_MONGODB_ENABLE_ADMIN="true"
+ENV ME_CONFIG_MONGODB_URL="mongodb://mongo:27017"
+ENV ME_CONFIG_SITE_SESSIONSECRET="secret"
+ENV VCAP_APP_HOST="0.0.0.0"
+
+RUN git clone https://github.com/mongo-express/mongo-express.git /app
```

Comment:
It looks like this is just building from master, can we build from a given NPM version? Otherwise maybe we could build from the tagged version with `-b <TAG>` and take that tag as a docker build argument? I think we can make the default master so it's easy for people to build locally.

Reply:
Testing with it hardcoded to v1.0.2 now. 

![image](https://github.com/mongo-express/mongo-express-docker/assets/3138467/63c80015-3268-4644-8c7e-24263145a029)

Reply:
Thanks for this PR. This is working for me if we use the tagged version of mongo-express. Would it be okay if I pushed a commit that uses your dockerfile changes in with the matrix pattern that was in place previously?

Reply:
Thanks! Just pushed with an option to override the version:

```bash
docker build --build-arg MONGO_EXPRESS_VERSION=v1.0.2 --tag mongo-express .
```

Reply:
> Would it be okay if I pushed a commit that uses your dockerfile changes in with the matrix pattern that was in place previously?

Sure! I'm not sure how it was working before so I may not be any help to you. But if you wanted to setup a github action that does the matrix builds that way and then publishes to Docker Hub, that could be another option.

Reply:
The official docker images are currently built from https://github.com/docker-library/official-images, they just need the git ref and the dockerfile for each version. It looks like this: 

```
# this file is generated via https://github.com/mongo-express/mongo-express-docker/blob/0b9b935b188caa945834a27f7475a059c4e60b13/generate-stackbrew-library.sh

Maintainers: Nick Cox <nickcox1008@gmail.com> (@knickers),
             John Steel <john@jskw.dev> (@BlackthornYugen)
GitRepo: https://github.com/mongo-express/mongo-express-docker.git
GitCommit: 0b9b935b188caa945834a27f7475a059c4e60b13

Tags: 1.0.0-20-alpine3.18, 1.0-20-alpine3.18, 1-20-alpine3.18
Architectures: amd64, arm64v8
GitCommit: 5fb537b2fd181e458f58137fdb55d061a8c23da5
Directory: 1.0/20-alpine3.18

Tags: 1.0.0-20, 1.0-20, 1-20, 1.0.0-20-alpine3.17, 1.0-20-alpine3.17, 1-20-alpine3.17
Architectures: amd64, arm64v8
GitCommit: 5fb537b2fd181e458f58137fdb55d061a8c23da5
Directory: 1.0/20-alpine3.17

Tags: 1.0.0-18-alpine3.18, 1.0-18-alpine3.18, 1-18-alpine3.18
Architectures: amd64, arm64v8
GitCommit: 5fb537b2fd181e458f58137fdb55d061a8c23da5
Directory: 1.0/18-alpine3.18

Tags: 1.0.0, 1.0, 1, 1.0.0-18, 1.0-18, 1-18, 1.0.0-18-alpine3.17, 1.0-18-alpine3.17, 1-18-alpine3.17, latest
Architectures: amd64, arm64v8
GitCommit: 5fb537b2fd181e458f58137fdb55d061a8c23da5
Directory: 1.0/18-alpine3.17
```

See:  https://github.com/docker-library/official-images/blob/master/library/mongo-express

Reply:
I thought we had stuff in the README.md that explained how the templates work. But I don't see it. I'll make a separate commit that makes that a bit more clear. Sorry that it's a bit confusing at the moment.

I can push something in a couple hours that should work. Need to cook dinner 🙂